### PR TITLE
[CI] Set default value for CI_NUM_EXECUTORS

### PR DESCRIPTION
**Pull request by lhutton1**
`task_cpp_unittest.sh` relies on `CI_NUM_EXECUTORS` being set as an environment variable, which causes the script to fail if the value is not set. To prevent this, setting the default as 1.

cc @driazati @leandron @areusch

### CODE REVIEW

File: tests/scripts/task_cpp_unittest.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ export OMP_NUM_THREADS=1
 
 # Build cpptest suite
 python3 tests/scripts/task_build.py \
-    --num-executors "${CI_NUM_EXECUTORS}" \
+    --num-executors "${CI_NUM_EXECUTORS:-1}" \
```
Comment (driazati) [inline tests/scripts/task_cpp_unittest.sh:35]: actually could you only use the default if the `CI` env variable isn’t set? just so we don’t end up triggering this behavior in CI where num executors should always be set. it also might be easier to put the logic in `task_build.py` and remove the `num-executors` flag

**Review by driazati — CHANGES_REQUESTED** [review]

File: tests/scripts/task_cpp_unittest.sh (hunk 1 above)
Comment (lhutton1) [reply tests/scripts/task_cpp_unittest.sh]: Sure, see what you think to the latest commit :)

**Review by driazati — APPROVED** [review]
nice! Since the CI will not read your `Jenkinsfile` (hence the errors as if you hadn't changed it), I pushed #10660 and the CI job there is reporting statuses to both PRs (since it's testing the same commit)

**Review by leandron — APPROVED** [review]
Thanks @lhutton1 @driazati !

**Pull request merged**